tasks/update_go.py

```diff
@@ -45,11 +63,11 @@ def update_go(
         raise exceptions.Exit(f"The version {version} isn't valid.")
 
     current_version = _get_repo_go_version()
-    current_major = _get_major_version(current_version)
-    new_major = _get_major_version(version)
+    current_major = _get_minor_version(current_version)
+    new_minor = _get_minor_version(version)
 
-    major_update = current_major != new_major
-    if major_update:
+    minor_update = current_major != new_minor
+    if minor_update:
         print(color_message("WARNING: this is a change of major version\n", "orange"))
```

Comment:
Yes definitely, nice catch
For the message itself I'm not sure as it can't be a change of major version, I think it's fair to just say minor (until Go has a v2 at least ;) )